### PR TITLE
feat(octavia): add floating ip to creation page

### PR DESCRIPTION
**Pull request by qpavy**
ref: MANAGER-12371

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/octavia-batch-1`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-12371
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- ~~[ ] Breaking change is mentioned in relevant commits~~

### CODE REVIEW

File: packages/manager/modules/octavia-load-balancer/src/create/constants.js [hunk 1]
```diff
@@ -107,10 +107,21 @@ export const AGORA_ADDON_FAMILY = 'octavia-loadbalancer';
 
 export const SIZE_FLAVOUR_REGEX = /octavia-loadbalancer.loadbalancer-([sml]).hour.consumption/;
 export const AGORA_GATEWAY_REGEX = /gateway.s.hour.consumption/;
+export const AGORA_FLOATINGIP_REGEX = /floatingip.floatingip.hour.consumption/;
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/create/constants.js:110]:
```suggestion
export const AGORA_FLOATING_IP_REGEX = /floatingip.floatingip.hour.consumption/;
```

File: packages/manager/modules/octavia-load-balancer/src/create/constants.js [hunk 2]
```diff
@@ -132,7 +143,10 @@ export default {
   TRACKING_NAME,
   SIZE_FLAVOUR_REGEX,
   AGORA_GATEWAY_REGEX,
+  AGORA_FLOATINGIP_REGEX,
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/create/constants.js:146]:
```suggestion
  AGORA_FLOATING_IP_REGEX,
```

File: packages/manager/modules/octavia-load-balancer/src/create/routing.js [hunk 3]
```diff
@@ -3,6 +3,7 @@ import {
   SIZE_FLAVOUR_REGEX,
   AGORA_ADDON_FAMILY,
   AGORA_GATEWAY_REGEX,
+  AGORA_FLOATINGIP_REGEX,
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/create/routing.js:6]:
```suggestion
  AGORA_FLOATING_IP_REGEX,
```

File: packages/manager/modules/octavia-load-balancer/src/create/routing.js [hunk 4]
```diff
@@ -42,6 +43,10 @@ export default /* @ngInject */ ($stateProvider) => {
         catalog.addons.filter((addon) =>
           addon.planCode.match(AGORA_GATEWAY_REGEX),
         )[0],
+      catalogFloatingIp: /* @ngInject */ (catalog) =>
+        catalog.addons.filter((addon) =>
+          addon.planCode.match(AGORA_FLOATINGIP_REGEX),
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/create/routing.js:48]:
```suggestion
          addon.planCode.match(AGORA_FLOATING_IP_REGEX),
```

File: packages/manager/modules/octavia-load-balancer/src/create/service.js [hunk 5]
```diff
@@ -25,18 +29,48 @@ export default class OctaviaLoadBalancerCreateService {
...
+          ),
+        });
+
+        availableFloatingIPs.unshift({
+          ip: null,
+          type: FLOATING_IP_TYPE.CREATE,
+          displayName: this.$translate.instant(
+            'octavia_load_balancer_create_floating_ip_field_new_floating_ip',
+          ),
+        });
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/create/service.js:71]: **nitpick:** Keep service for only fetching data from server. Move this logic to controller to make this function reusable.

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/octavia-load-balancer/src/create/translations/Messages_fr_FR.json [hunk 6]
```diff
@@ -12,8 +12,17 @@
   "octavia_load_balancer_create_region_title": "Sélectionnez une région",
   "octavia_load_balancer_create_region_intro": "Veuillez sélectionner la région Public Cloud où votre Load Balancer sera hébergé. Parmi les regions public cloud où un réseau privé est défini, sélectionnez celle où le Load Balancer sera hébergé. Vous ne trouvez pas la région que vous désirez dans la liste ? Pour en savoir plus sur la disponibilité des fonctions par région, ",
   "octavia_load_balancer_create_region_link": "Cliquez ici",
+  "octavia_load_balancer_create_floating_ip_title": "Attacher une IP Publique",
+  "octavia_load_balancer_create_floating_ip_intro": "Pour recevoir du trafic public, vous devez attacher une Floating IP à votre Load Balancer",
+  "octavia_load_balancer_create_floating_ip_field": "Attacher une IP Publique (floating IP)",
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/create/translations/Messages_fr_FR.json:17]:
```suggestion
  "octavia_load_balancer_create_floating_ip_field": "Attacher une IP Publique (Floating IP)",
```

File: packages/manager/modules/octavia-load-balancer/src/create/translations/Messages_fr_FR.json [hunk 7]
```diff
@@ -12,8 +12,17 @@
...
+  "octavia_load_balancer_create_floating_ip_field_no_floating_ip": "Aucune IP Publique",
+  "octavia_load_balancer_create_floating_ip_field_new_floating_ip": "Nouvelle IP Publique",
+  "octavia_load_balancer_create_floating_ip_new_information": "La création de votre Floating IP se fera à la fin du processus de configuration.",
+  "octavia_load_balancer_create_floating_ip_new_price": "La nouvelle floating IP sera facturée {{price}}",
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/create/translations/Messages_fr_FR.json:21]:
```suggestion
  "octavia_load_balancer_create_floating_ip_new_price": "La nouvelle Floating IP sera facturée {{price}}",
```

File: packages/manager/modules/octavia-load-balancer/src/create/constants.js [hunk 8]
```diff
@@ -107,10 +107,21 @@ export const AGORA_ADDON_FAMILY = 'octavia-loadbalancer';
...
 export const NETWORK_PRIVATE_VISIBILITY = 'private';
 
+export const FLOATING_IP_TYPE = {
+  NO_IP: 'none',
+  CREATE: 'create',
+  IP: 'ip',
+};
+
+export const FLOATING_IP_CREATE_DESCRIPTION =
+  'FIP created by OVHCloud Control Panel (Manager) for Load Balancer';
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/create/constants.js:123]: **question:** No translation required for this?

File: packages/manager/modules/octavia-load-balancer/src/create/template.html [hunk 9]
```diff
@@ -90,13 +90,82 @@
...
+            data-ng-init="stepFloatingIp = $ctrl.stepper.loadBalancerFloatingIp"
+            data-name="{{:: stepFloatingIp.name}}"
+            data-header="{{:: 'octavia_load_balancer_create_floating_ip_title' | translate }}"
+            data-prevent-next="true"
+        >
+            <div class="mb-4">
+                <p
+                    data-translate="octavia_load_balancer_create_floating_ip_intro"
+                ></p>
+            </div>
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/create/template.html:104]: **nitpick:** Unnecessary `div` can be avoided
```suggestion
                <p
                    class="mb-4"
                    data-translate="octavia_load_balancer_create_floating_ip_intro"
                ></p>
```

File: packages/manager/modules/octavia-load-balancer/src/create/template.html [hunk 10]
```diff
@@ -90,13 +90,82 @@
...
+            <oui-spinner data-ng-if="$ctrl.floatingIpLoading"></oui-spinner>
+            <div class="container-fluid" data-ng-if="!$ctrl.floatingIpLoading">
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/create/template.html:106]: **nitpick:** Usage of `col-*` should be only within `row` wrapper class in Bootstrap 4.
```suggestion
            <div class="container-fluid row" data-ng-if="!$ctrl.floatingIpLoading">
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

**Review by anooparveti — CHANGES_REQUESTED** [review]
I see that we are handling the success & error  in the service https://github.com/ovh/manager/blob/aedef32215f60b446c506fedbbcc5741469ad67a/packages/manager/modules/octavia-load-balancer/src/create/service.js#L171-L190

Can we please move this to the controller?

File: packages/manager/modules/octavia-load-balancer/src/create/constants.js (hunk 8 above)
Comment (qpavy) [reply packages/manager/modules/octavia-load-balancer/src/create/constants.js:123]: Nop this is why I put this in constants

**Review by JacquesLarique — APPROVED** [review]

**Pull request merged**